tests/test_core/test_task.py

```diff
@@ -0,0 +1,237 @@
+import os
+import sys
+import logging
+
+from collections import OrderedDict
+
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+import setup_malcolm_paths
+
+import unittest
+from mock import MagicMock, call, ANY
+
+#module imports
+from malcolm.compat import queue
+from malcolm.core.task import Task
+from malcolm.core.response import Response
+from malcolm.core.request import Request
+from malcolm.core.method import Method
+from malcolm.core.future import Future, CancelledError, TimeoutError
+from malcolm.core.meta import Meta
+from malcolm.core.attribute import Attribute
+from malcolm.core.block import Block
+from malcolm.core.serializable import Serializable
+
+import logging
+logging.basicConfig(level=logging.DEBUG)
+
+class TestTask(unittest.TestCase):
+
+    def setUp(self):
+        self.callback_result = 0
+        self.callback_value = ''
+        meta = Meta("meta", "meta for unit tests")
+        self.block = MagicMock()
+        self.proc = MagicMock(q=queue.Queue())
+        self.proc.create_queue = MagicMock(return_value=queue.Queue())
```

Comment:
This might want to be side_effect=queue.Queue, otherwise every create_queue returns the same instance...